### ISSUE TITLE
Movement animations

### ISSUE THREAD

**Issue by Seeker04** [Seeker04/plwm#45]
Investigate the feasibility of simple animations, i.e., when windows are created, moved, resized, closed.

Rationale: compositors, like [picom](https://github.com/yshui/picom), usually don't implement animations.

There is a [picom fork](https://github.com/jonaburg/picom) that has animations, however it seems relatively unmaintained.

**Possible implementation:** (might be terrible performance wise)
Interpolate `X` and/or `Y` and/or `W` and/or `H` values in a configured timeframe making multiple `x_move_resize_window/6` calls instead of the single one we use now. Should be async, i.e., started on a new thread so plwm is not blocked.

**Comment by Seeker04** [comment]
**Note:** if we put the interpolation to a separate thread, then operations like switching workspace while the animation is playing could potentially result in an inconsistent display.

**Comment by Seeker04** [comment]
Tried several methods:

**1. Interpolate on detached threads while continuing to process X events**
Not feasible, because plwm can crash if I spam the close window hotkey while movement animations are not yet finished (a scenario not too realistic, but still, we must **never** crash). Setting XErrorHandler to a dummy and other similar safeguards (from dwm) can make the crash less frequent, but it can still occur. It is well-known that Xlib was not designed with such concurrency in mind. [XCB](https://github.com/Seeker04/plwm/issues/51) might solve this, but there's no guarantee.

**2. Don't use threads at all**
Looks bad when all windows move/resize one after another during relayouts. Also, `animation_time * no. windows` blocks (queues) input for too long.

**3. Block input for the time of animation, but parallel move/resize all windows with joined threads**
This gives the best of both worlds. Safe, yet good-looking, because all windows move at the same time. User input is only blocked for `1 * animation_time` which shouldn't be high in the first place (e.g. `0.2` seconds). Also, input received during the animation is still processed right after, because X queues it, so nothing is lost.

**tl;dr** We should go with no. 3, because it's safe and there is no real downside compared to no. 1. Issue raised in my previous comment is also non-existent with this solution.
